nh.c

```diff
@@ -461,3 +471,52 @@ void nh_failsafe_uninstall(nh_failsafe_t *fs) {
     nh_log("(NickelHook) ... failsafe: info: deleting %s", fs->tmp);
     unlink(fs->tmp);
 }
+
+// --- File/directory deletion
+
+// The following blacklist helps prevent any whoopsies when deleting files or directories during uninstall
+
+// Let's not allow deleting any files in /bin, /sbin, /etc/init.d or u-boot stuff.
+static const char *delete_prefix_blacklist[] = {
+    "/bin",
+    "/sbin",
+    // note, '/etc/u-boot/*/u-boot.recovery' is a critical file for recovery
+    "/etc/u-boot",
+    "/etc/init.d",
+    "/usr/local/Kobo/pickel",
+};
+
+// This is for the schmuck who forgot to terminate his array of paths
+// and accidentally deleted '/bin/sh' due to an out-of-bounds read.
+bool nh_delete_path(const char *path, bool is_dir) {
+    if (!path) {
+        nh_log("(NickelHook) no path supplied");
+        return false;
+    }
+    char canonical_path[PATH_MAX] = {0};
+    if (!realpath(path, canonical_path)) {
+        nh_log("(NickelHook) unable to get canonical path for %s : %m", path);
+        return false;
+    }
+    for (unsigned int i = 0; i < (sizeof(delete_prefix_blacklist) / sizeof(delete_prefix_blacklist[0])); i++) {
```

Comment:
I prefer using `size_t` over `unsigned int` and  `sizeof(*delete_prefix_blacklist)` over `sizeof(delete_prefix_blacklist[0])`.

Reply:
Umm don't know why I didn't do that in the first place 😊, since that's what sizeof 'returns'. Good catch.